stars-core/src/test/kotlin/tools/aqua/stars/core/hooks/MinTicksPerSegmentHookTest.kt

```diff
@@ -0,0 +1,186 @@
+          MinTicksPerSegmentHook(minTicks = 2, failPolicy = EvaluationHookResult.OK))
+
+      runEvaluation(writePlots = false, writePlotDataCSV = false)
+
+      assertEquals(2, segmentCountMetric.getState())
+    }
+  }
+
+  /** Test MinTicksPerSegmentHook with fail policy SKIP. */
+  @Test
+  fun `Test MinTicksPerSegmentHook with fail policy SKIP`() {
+    val tsc =
+        tsc<
+            SimpleEntity,
+            SimpleTickData,
+            SimpleSegment,
+            SimpleTickDataUnit,
+            SimpleTickDataDifference> {
+          any("") {}
+        }
+    TSCEvaluation(tsc = tsc, segments = segments()).apply {
+      val segmentCountMetric = setup()
+      registerPreSegmentEvaluationHooks(
+          MinTicksPerSegmentHook(minTicks = 2, failPolicy = EvaluationHookResult.SKIP))
+
+      runEvaluation(writePlots = false, writePlotDataCSV = false)
+
+      assertEquals(1, segmentCountMetric.getState())
+    }
+  }
+
+  /** Test MinTicksPerSegmentHook with fail policy ABORT. */
+  @Test
+  fun `Test MinTicksPerSegmentHook with fail policy ABORT`() {
+    val tsc =
+        tsc<
+            SimpleEntity,
+            SimpleTickData,
+            SimpleSegment,
+            SimpleTickDataUnit,
+            SimpleTickDataDifference> {
+          any("") {}
+        }
+    TSCEvaluation(tsc = tsc, segments = segments()).apply {
+      setup()
+      registerPreSegmentEvaluationHooks(
+          MinTicksPerSegmentHook(minTicks = 2, failPolicy = EvaluationHookResult.ABORT))
+
+      assertFailsWith<PreSegmentEvaluationHookAbort> {
+        runEvaluation(writePlots = false, writePlotDataCSV = false)
+      }
+    }
+  }
+
+  /** Test MinTicksPerSegmentHook with #ticks less than or equal to minTicks. */
+  @Test
+  fun `Test MinTicksPerSegmentHook with #ticks less than or equal to minTicks`() {
+    val tsc =
+        tsc<
+            SimpleEntity,
+            SimpleTickData,
+            SimpleSegment,
+            SimpleTickDataUnit,
+            SimpleTickDataDifference> {
+          any("") {}
+        }
+    TSCEvaluation(tsc = tsc, segments = segments()).apply {
+      val segmentCountMetric = setup()
+      registerPreSegmentEvaluationHooks(
+          MinTicksPerSegmentHook(minTicks = 1, failPolicy = EvaluationHookResult.ABORT))
+
+      runEvaluation(writePlots = false, writePlotDataCSV = false)
+
+      assertEquals(2, segmentCountMetric.getState())
+    }
+  }
+
+  /** Test MinTicksPerSegmentHook with #ticks negative. */
+  @Test
+  fun `Test MinTicksPerSegmentHook with #ticks negative`() {
+    assertFailsWith<IllegalArgumentException> {
+      MinTicksPerSegmentHook<
+          SimpleEntity,
+          SimpleTickData,
+          SimpleSegment,
+          SimpleTickDataUnit,
+          SimpleTickDataDifference>(
+          minTicks = -1)
+    }
+  }
+
+  private fun TSCEvaluation<
+      SimpleEntity, SimpleTickData, SimpleSegment, SimpleTickDataUnit, SimpleTickDataDifference>
+      .setup():
```

Comment:
This could be extracted to a util class? It seems its the same as for the NodesInTSCHook